notion-server/src/eth_magic.rs

```diff
@@ -0,0 +1,165 @@
+use std::default;
+use std::collections::{HashMap};
+
+#[derive(Debug)]
+pub struct EthMagicManager {
+    beacon_chain: BeaconChain,
+    shard_chains: Vec<ShardChain>,
+}
+
+impl EthMagicManager {
+    pub fn new() -> Self {
+        Self {
+            beacon_chain: BeaconChain::new(),
+            shard_chains: Vec::new(),
+        }
+    }
+
+    // Creates a new execution environment on the BeaconChain and returns the
+    // index of the created execution environment
+    pub fn add_new_execution_environment(&mut self, ee: CreateEeArgs) -> u32 {
+        let execution_environment = ExecutionEnvironment {
+            wasm_code: ee.wasm_code,
+        };
+        self.beacon_chain.execution_environments.push(execution_environment);
```

Comment:
`BeaconChain` provides a function that handles adding an execution environment. Should that be called here?

Reply:
Good catch -- I agree it should be one or the other, but not both.  I'll make this consistent.